pvlib/tests/bifacial/test_utils.py

```diff
@@ -86,3 +88,73 @@ def test__vf_ground_sky_2d(test_system_fixed_tilt):
     vf = utils._vf_ground_sky_2d(pts[0], ts['rotation'], ts['gcr'],
                                  ts['pitch'], ts['height'], max_rows=1)
     assert np.isclose(vf, vfs_gnd_sky[0])
+
+
+def test_vf_row_sky_2d(test_system_fixed_tilt):
+    ts, _, _ = test_system_fixed_tilt
+    # with float input, fx at top of row
+    vf = utils.vf_row_sky_2d(1., ts['gcr'], ts['surface_tilt'])
+    expected = 0.5 * (1 + cosd(ts['surface_tilt']))
+    assert np.isclose(vf, expected)
+    # with array input
+    fx = np.array([0., 0.5, 1.])
+    vf = utils.vf_row_sky_2d(fx, ts['gcr'], ts['surface_tilt'])
+    phi = masking_angle(ts['surface_tilt'], ts['gcr'], fx)
+    expected = 0.5 * (1 + cosd(ts['surface_tilt'] + phi))
+    assert np.allclose(vf, expected)
+
+
+def test_vf_row_sky_2d_integ(test_system_fixed_tilt):
+    ts, _, _ = test_system_fixed_tilt
+    # with float input, check end position
+    vf = utils.vf_row_sky_2d_integ(1., 1., ts['gcr'], ts['surface_tilt'])
+    expected = utils.vf_row_sky_2d(1., ts['gcr'], ts['surface_tilt'])
+    assert np.isclose(vf, expected)
+    # with array input
+    fx0 = np.array([0., 0.5])
+    fx1 = np.array([0., 0.8])
+    vf = utils.vf_row_sky_2d_integ(fx0, fx1, ts['gcr'],
+                                   ts['surface_tilt'])
+    phi = masking_angle(ts['surface_tilt'], ts['gcr'], fx0[0])
+    y0 = 0.5 * (1 + cosd(ts['surface_tilt'] + phi))
+    x = np.arange(fx0[1], fx1[1], 1e-4)
+    phi_y = masking_angle(ts['surface_tilt'], ts['gcr'], x)
+    y = 0.5 * (1 + cosd(ts['surface_tilt'] + phi_y))
+    y1 = np.trapz(y, x) / (fx1[1] - fx0[1])
+    expected = np.array([y0, y1])
+    assert np.allclose(vf, expected, rtol=1e-3)
+
+
+def test_vf_row_ground_2d(test_system_fixed_tilt):
+    ts, _, _ = test_system_fixed_tilt
+    # with float input, fx at bottom of row
+    vf = utils.vf_row_ground_2d(0., ts['gcr'], ts['surface_tilt'])
+    expected = 0.5 * (1. - cosd(ts['surface_tilt']))
+    assert np.isclose(vf, expected)
+    # with array input
+    fx = np.array([0., 0.5, 1.0])
+    vf = utils.vf_row_ground_2d(fx, ts['gcr'], ts['surface_tilt'])
+    phi = infinite_sheds._ground_angle(fx, ts['surface_tilt'], ts['gcr'])
```

Comment:
Makes me wonder if `infinite_sheds._ground_angle` ought to sit next to `shading.masking_angle`.  Not needed for this PR of course.